web/modules/custom/joinup_communities/tallin/src/Plugin/Field/FieldFormatter/TallinEntryFormatter.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Drupal\tallin\Plugin\Field\FieldFormatter;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\FormatterBase;
+use Drupal\Core\TypedData\TypedDataInterface;
+use Drupal\Core\Url;
+use Drupal\tallin\Plugin\Field\FieldType\TallinEntryItem;
+
+/**
+ * Plugin implementation of the 'tallin_entry' formatter.
+ *
+ * @FieldFormatter(
+ *   id = "tallin_entry",
+ *   label = @Translation("Complex tallin formatter"),
+ *   field_types = {
+ *     "tallin_entry"
+ *   }
+ * )
+ */
+class TallinEntryFormatter extends FormatterBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    // All fields are single-value fields.
+    $item = $items->first();
+    if (empty($item)) {
+      return [];
+    }
+
+    $color = $this->getOptionToColor($item);
+    $option = $this->getOptionToString($item);
+    $font_color = in_array($color, ['red', 'green', 'grey']) ? 'white' : 'black';
+    $build = [
+      '#theme' => 'tallin_entry',
+      '#title' => $this->fieldDefinition->getLabel(),
+      '#status' => [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => $option,
+        '#attributes' => [
+          'style' => "background-color: {$color}; color: {$font_color}",
```

Comment:
Normally, Drupal discorages the inline CSS. Using CSS classes would have been better.

Reply:
Yes indeed CSS classes and avoid all the usages of 'html_tag'. Use a render element if needed or a template for the rendering of all this element.